jmh-core/src/main/java/org/openjdk/jmh/runner/CompilerHints.java

```diff
@@ -188,10 +228,12 @@ public static List<String> getCompileCommandFiles(List<String> command){
      * @param command all -XX:CompileCommandLine args will be removed and a merged file will be set
      */
     public static void addCompilerHints(List<String> command) {
-        if (!isHintCompatibleVM()) {
-            System.err.println("WARNING: Not a HotSpot compiler command compatible VM (\""
-                    + System.getProperty("java.vm.name") + "-" + System.getProperty("java.version")
-                    + "\"), compilerHints are disabled.");
+        if (!compilerHintsEnabled()) {
+            if (compilerHintsSelect() == CompilerHintsSelect.AUTO) {
+                System.err.println("WARNING: Not a HotSpot compiler command compatible VM (\""
+                        + System.getProperty("java.vm.name") + "-" + System.getProperty("java.version")
+                        + "\"), compilerHints are disabled.");
```

Comment:
While you are at it, please change "compilerHints" -> "compiler hints".

Reply:
done